### PR TITLE
fix: fix Contentful sync limit

### LINKED ISSUE #21026: gatsby-source-contentful Response Size Too Big
## Description
We're running into an issue similar to #20081 with gatsby-source-contentful where once we have too many entries, we're getting an error "400 Response size too big. Maximum allowed response size: 7340032b." I've run into the same error with contentful CLI which can be mitigated with --max-allowed-limit, but there doesn't seem to be any pagination limit options in gatsby-source-contentful.

### Steps to reproduce

Run gatsby develop or gatsby build

### Expected result
The site should build

### Actual result
Error: Request failed with status code 400

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

  System:
    OS: macOS 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Shell: 3.0.2 - /usr/local/bin/fish
  Binaries:
    Node: 13.1.0 - /usr/local/bin/node
    Yarn: 1.19.1 - /usr/local/bin/yarn
    npm: 6.12.1 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/local/bin/python
  Browsers:
    Chrome: 79.0.3945.130
    Firefox: 70.0.1
    Safari: 13.0.5
  npmPackages:
    gatsby: ^2.19.1 => 2.19.1
    gatsby-cli: ^2.8.2 => 2.8.19
    gatsby-image: ^2.1.4 => 2.2.36
    gatsby-plugin-canonical-urls: ^2.0.12 => 2.1.18
    gatsby-plugin-catch-links: ^2.0.13 => 2.1.21
    gatsby-plugin-feed: ^2.2.3 => 2.3.25
    gatsby-plugin-google-tagmanager: ^2.1.24 => 2.1.24
    gatsby-plugin-i18n: ^1.0.1 => 1.0.1
    gatsby-plugin-manifest: ^2.2.33 => 2.2.33
    gatsby-plugin-netlify: ^2.0.15 => 2.1.30
    gatsby-plugin-offline: ^2.1.3 => 2.2.10
    gatsby-plugin-react-helmet: ^3.0.12 => 3.1.18
    gatsby-plugin-s3: ^0.3.2 => 0.3.2
    gatsby-plugin-schema-snapshot: ^1.0.0 => 1.0.0
    gatsby-plugin-sharp: ^2.1.5 => 2.3.8
    gatsby-plugin-sitemap: ^2.0.12 => 2.2.24
    gatsby-plugin-styled-components: ^3.0.7 => 3.1.16
    gatsby-remark-autolink-headers: ^2.0.16 => 2.1.21
    gatsby-remark-images-contentful: ^2.0.10 => 2.1.27
    gatsby-remark-responsive-iframe: ^2.2.10 => 2.2.30
    gatsby-source-contentful: ^2.1.55 => 2.1.70
    gatsby-source-filesystem: ^2.0.39 => 2.1.42
    gatsby-transformer-remark: ^2.3.10 => 2.6.42
    gatsby-transformer-sharp: ^2.1.21 => 2.3.9
  npmGlobalPackages:
    gatsby-cli: 2.7.30

### PR DESCRIPTION
**Pull request by Khaledgarbaya**
## Description
Due to some technical limitation the response payload should not be greater than 7MB when pulling content from Contentful.
This can cause the a problem for some users if they have entries with a lot of text in them. To fix this my Team introduced a new option called `limit` on the Contentful sync API to allow user to break their payload into smaller chunks and avoid the 400 error.

### Documentation
```
// In your gatsby-config.js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-source-contentful`,
      options: {
        spaceId: `your_space_id`,
        accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
        limit: 50 // default is 100
      },
    },
  ],
}
```

## Related Issues
Fixes #21026

### CODE REVIEW

**Comment by jessepinho** [comment]
There's actually a `pageLimit` parameter already, but it's only used for determining how many content types to fetch at a time (??). Not sure if that should be used instead of this, but just wanted to make sure you know about it @kgar in case that's a better option

**Comment by pvdz** [comment]
Thanks, that's nice :dagger: 
Could you update the docs for this?

**Comment by Khaledgarbaya** [comment]
@pvdz I switched to using the existing param `pageLimit` which already documented and tested so I think we don't need to add any more docs

**Comment by pvdz** [comment]
Thank you. I think it would be helpful to have the actual limits, as you described them here, in that doc as well? Or do you think users generally not care about this.

> Due to some technical limitation the response payload should not be greater than 7MB when pulling content from Contentful.

**Comment by laurieontech** [comment]
In terms of the change it feels reasonable to document the new default. I struggle a bit with whether to add the additional context. While it's relevant, it's technically a contentful API implementation detail and it only pops up if the user makes an active choice to override the default. In which case we have to hope they know what they're doing?

File: packages/gatsby-source-contentful/README.md [hunk 1]
```diff
@@ -168,9 +168,9 @@ Using the ID is a much more stable property to work with as it will change less
 
 If you are confident your Content Types will have natural-language IDs (e.g. `blogPost`), then you should set this option to `false`. If you are unable to ensure this, then you should leave this option set to `true` (the default).
 
-**`pageLimit`** [number][optional] [default: `1000`]
+**`pageLimit`** [number][optional] [default: `100`]
 
-Number of entries to retrieve from Contentful at a time. This can be adjusted to fix issues related to "Response size too big" error.
+Number of entries to retrieve from Contentful at a time. Due to some technical limitation the response payload should not be greater than 7MB when pulling content from Contentful. If you encounter this issue can set this param to a lower number than 100, e.g `50`.
```
Comment (laurieontech) [inline packages/gatsby-source-contentful/README.md]:
```suggestion
Number of entries to retrieve from Contentful at a time. Due to some technical limitations, the response payload should not be greater than 7MB when pulling content from Contentful. If you encounter this issue you can set this param to a lower number than 100, e.g `50`.
```

**Comment by laurieontech** [comment]
Looks good! I'm gonna get this in.

**Comment by jessepinho** [comment]
Looks like this doesn't work until the `contentful` NPM module is re-published, and `package.json` is updated to use the new `contentful`. Here's the deal:
- This PR sends a `limit` parameter to the `client.sync()` method. 
- The `sync` method takes things from there, requesting items from the Contentful Sync API in pages that are sized via the `limit` parameter. NOTE: Since the `limit` parameter was passed in as part of the query params object, the `client.sync()` method keeps sending the `limit` param on each additional request. This is despite the fact that this PR [only sends the `limit` param if there's no sync token](https://github.com/gatsbyjs/gatsby/pull/21644/files#diff-6e25753afbe27b5e9223708cc884666fR81). The reason is that the object in the line linked above is the only one that ever gets passed to `client.sync()` — you don't need to call `client.sync` multiple times, since it takes care of pagination itself.
- The Sync API succeeds on the first request, and then issues a 400 on the second request, complaining that you can't include both a `limit` and a `sync_token` in the same request.

Note that this issue is [taken care of](https://github.com/contentful/contentful.js/commit/10dfe9e0a7e85c4761055df1a043e7bd58eae3f1#diff-e10b87596793e37b345e154f12119ce6R143) in the _latest master_ of the Contentful JS SDK — for any request that includes a sync token, the `limit` param is deleted, so as to avoid this issue. However, the latest master is not published as an NPM package yet. And when it is, it still needs to be referenced by `gatsby-source-contentful`'s `package.json`.

In short, to fix this issue:
- A new version of `contentful` needs to be published to NPM (taking in the latest changes from `master`).
- `gatsby-source-contentful` needs to be updated to use the newly published version of `contentful`.
- `gatsby-source-contentful` then needs to be re-published. 

(Optionally, you could also get rid of the logic that checks for the sync token [here](https://github.com/gatsbyjs/gatsby/pull/21644/files#diff-6e25753afbe27b5e9223708cc884666fR79-R81), since it's really not being used.)

cc @Khaledgarbaya

**Comment by Khaledgarbaya** [comment]
@jessepinho according to this comment https://github.com/contentful/contentful.js/pull/373#issuecomment-589629616 the PR should be included in the new version.
that version was added in this [PR](https://github.com/gatsbyjs/gatsby/pull/21644/files#diff-1ddcbc06ecb2e276e3bdd6111000427aR16)

and looking at the new version on [unpkg](https://unpkg.com/browse/contentful@7.14.0/dist/contentful.node.js) the line of code is there.

could this be a `lock` file issue?

**Comment by jessepinho** [comment]
@Khaledgarbaya ahh sorry! I had simply copied the new built files into my `node_modules` for testing, and didn't also update `contentful`. Sorry about that 🤦‍♂ looks like it's working after all.

How do we get this published?

**Comment by pvdz** [comment]
Fwiw, the package was just published at gatsby-source-contentful@2.1.87

**Comment by jessepinho** [comment]
Great, thanks @pvdz !
